### PR TITLE
refactor(protocol-engine): create `ActionDispatcher` action pipeline

### PR DESCRIPTION
**Pull request by mcous**
## Overview

This is an intermediate PR to address #8374. In order to implement 8374, we'll be adding a rudimentary plugin interface to the engine to allow other objects to register something that will react to actions as they flow through the engine's internal action pipeline.

This PR makes that internal action pipeline an explicit collaborator structure: the `ActionDispatcher`. In future PRs, the `ActionDispatcher` will gain the ability to insert action handlers into the flow. For now, it contains a single destination: the `StateStore`.

## Changelog

- Create `ActionDispatcher` interface to push actions to the `StateStore`
- Replace all "public" calls to `state_store.handle_action` with `action_dispatcher.dispatch`

## Review requests

Unit and automated smoke tests are all passing, and there will be one or two more PRs in this line, so focus on code and structure on this one. Robot testing will be _much_ easier at the end, and we can address any behavior bugs there

## Risk assessment

Low, feature flagged and in active development

### CODE REVIEW

File: api/src/opentrons/protocol_engine/actions/action_dispatcher.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+"""Action pipeline module."""
+from .action_handler import ActionHandler
+from .actions import Action
+
+
+class ActionDispatcher:
+    """A pipeline, with an endpoint, that actions cna be dispatched into."""
```
Comment (SyntaxColoring) [inline api/src/opentrons/protocol_engine/actions/action_dispatcher.py]:
```suggestion
    """A pipeline, with an endpoint, that actions can be dispatched into."""
```

File: api/src/opentrons/protocol_engine/protocol_engine.py [hunk 2]
```diff
@@ -45,29 +46,40 @@ def __init__(
...
 
     @property
     def state_view(self) -> StateView:
         """Get an interface to retrieve calculated state values."""
         return self._state_store
 
+    # def add_plugin(self, plugin: AbstractPlugin) -> None:
+    #     """Add a plugin to the engine to customize behavior."""
+    #     raise NotImplementedError("add_plugin not yet implemented")
+
```
Comment (SyntaxColoring) [inline api/src/opentrons/protocol_engine/protocol_engine.py]: Was this intended to be uncommented or removed?

**Review by SyntaxColoring — APPROVED** [review]
Makes sense so far.

File: api/src/opentrons/protocol_engine/protocol_engine.py (hunk 2 above)
Comment (mcous) [reply api/src/opentrons/protocol_engine/protocol_engine.py]: Oh, oops. Yeah meant to remove this

**Pull request merged**